clang/include/clang/Basic/LangOptions.h

```diff
@@ -1096,6 +1100,177 @@ inline void FPOptions::applyChanges(FPOptionsOverride FPO) {
   *this = FPO.applyOverrides(*this);
 }
 
+/// Atomic control options
+class AtomicOptionsOverride;
+class AtomicOptions {
+public:
+  using storage_type = uint16_t;
+
+  static constexpr unsigned StorageBitSize = 8 * sizeof(storage_type);
+
+  static constexpr storage_type FirstShift = 0, FirstWidth = 0;
+#define OPTION(NAME, STR, TYPE, WIDTH, PREVIOUS)                               \
```

Comment:
This `PREVIOUS` is incredibly error-prone, and IMO, something we shouldn't do (at least manually).  I'd perhaps be ok with tablegen for this one, but frankly, I'm not convinced it is necessary.

Reply:
will remove the macro-based design